### PR TITLE
Add comprehensive tests for OidcClientUriBuilder

### PR DESCRIPTION
**Pull request by onkar717**
### Summary

Adds additional unit tests for `OidcClientUriBuilder` to improve test coverage of its public API.

### Isssue

https://redhat.atlassian.net/browse/ELY-3019

### Details

The existing `OidcClientUriBuilderTest` contained only a minimal baseline test. This change expands the test suite to cover:

* Factory methods (`fromUri`, `fromPath`, `fromTemplate`)
* URI construction and build methods
* Path handling
* Query parameter handling
* Fragment and user info handling
* Matrix parameters
* Getter methods
* clone() behavior
* Static utility methods
* Error handling and edge cases
* Integration scenarios

### Notes

* Only test code was added
* Existing test behavior preserved
* Tests follow existing project patterns (JUnit 4)

### Testing

COmmand to execute test - 

```
mvn test -pl http/oidc -Dtest=OidcClientUriBuilderTest
```

All tests pass successfully.

### CODE REVIEW

**Comment by skyllarr** [comment]
@onkar717 Please add jira issue number on the beginning of the commit. Thank you

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcClientUriBuilderTest.java [hunk 1]
```diff
@@ -22,15 +22,30 @@
...
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
 
 /**
+ * Tests for {@link OidcClientUriBuilder}.
+ *
+ * <p>The original {@link #testKeycloakUriBuilder()} method validates baseline
```
Comment (skyllarr) [inline http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcClientUriBuilderTest.java]: The comments should be relevant in the future as well. There will be no original method. Please change to something more generic that describes the test class

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcClientUriBuilderTest.java [hunk 2]
```diff
@@ -22,15 +22,30 @@
...
+ * {@link KeycloakUriBuilder} behavior. All remaining methods exercise the
+ * {@code OidcClientUriBuilder} public API directly.</p>
+ *
  * @author rmartinc
  */
 public class OidcClientUriBuilderTest {
 
+    // ------------------------------------------------------------------ //
+    //  Existing KeycloakUriBuilder baseline tests (unchanged)             //
```
Comment (skyllarr) [inline http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcClientUriBuilderTest.java]: same as above, either remove this comment or generically describe what the method does

**Comment by skyllarr** [comment]
@onkar717 Please squash the 2 commits together and remove the empty file named `,` that was added

**Review by skyllarr — CHANGES_REQUESTED** [review]

**Review by skyllarr — APPROVED** [review]
